### PR TITLE
Select dayoff type dialog switched to radio buttons

### PR DESCRIPTION
**Pull request by DesmanLead**

### CODE REVIEW

**Comment by DesmanLead** [comment]
simulator screen shot - iphone se - 2018-11-29 at 20 02 47

File: mobile/src/common/radio-buttons-group.component.tsx [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+                    ))}
+                </View>
+            </View>
+        );
+    }
+
+    private validate(buttons: RadioButton[]) {
+        let selected = false; // Variable to check if "selected: true" for more than one button.
+        buttons.map(button => {
+            button.color = button.color ? button.color : '#444';
```
Comment (serjKim) [inline mobile/src/common/radio-buttons-group.component.tsx]: props shouldn't be mutated

File: mobile/src/common/radio-buttons-group.component.tsx [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+    public render() {
+        const opacity = this.props.disabled ? 0.2 : 1;
+
+        let layout: ViewStyle = { flexDirection: 'row' };
+        let margin: ViewStyle = { marginLeft: 10 };
+        if (this.props.layout === 'column') {
+            layout = { alignItems: 'center' };
+            margin = { marginTop: 10 };
+        }
+        let textStyle: TextStyle = {
```
Comment (serjKim) [inline mobile/src/common/radio-buttons-group.component.tsx]: const textStyle = StyleSheet.flatten([
...
]);
all styles should use StyleSheet.flatten/create, it optimizes the passing ones through the bridge

Comment (antonm76) [reply mobile/src/common/radio-buttons-group.component.tsx]: Flattening ruins optimization.
https://github.com/robinpowered/glamorous-native/issues/5
"By using StyleSheet.flatten, we no longer get the performance advantage from registered styles."
https://github.com/facebook/react-native/blob/af949877e67c9ff56089436c7980037a9e332c7c/Libraries/StyleSheet/StyleSheet.js#L201
https://github.com/facebook/react-native/blob/af949877e67c9ff56089436c7980037a9e332c7c/Libraries/StyleSheet/flattenStyle.js#L21

File: mobile/src/common/radio-buttons-group.component.tsx (hunk 1 above)
Comment (antonm76) [reply mobile/src/common/radio-buttons-group.component.tsx]: Not anymore. 3ffda78b754b078f3ac42b00401ac6da16609321

File: mobile/src/common/radio-buttons-group.component.tsx [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+
+//============================================================================
+export interface RadioButton {
+    selectionIndex: number;
+    label: string;
+
+    labelStyle?: TextStyle;
+    color?: string;
+    disabled?: boolean;
+    layout?: string;
```
Comment (serjKim) [inline mobile/src/common/radio-buttons-group.component.tsx]: layout?: 'column' | 'row'

File: mobile/src/common/radio-buttons-group.component.tsx [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+    size?: number;
+}
+
+//============================================================================
+interface RadioButtonProps extends RadioButton {
+    onPress: (label: string) => void;
+}
+
+//============================================================================
+class RadioButtonComponent extends React.Component<RadioButtonProps> {
```
Comment (serjKim) [inline mobile/src/common/radio-buttons-group.component.tsx]: just `RadioButton`

Comment (antonm76) [reply mobile/src/common/radio-buttons-group.component.tsx]: there is already interface with this name

**Pull request merged**